gutils/gimagereadpng.c

```diff
@@ -91,11 +92,12 @@ return( NULL );
     }
 
 #if (PNG_LIBPNG_VER < 10500)
-    if (setjmp(png_ptr->jmpbuf))
+    test = setjmp(png_ptr->jmpbuf);
 #else
-    if (setjmp(*png_set_longjmp_fn(png_ptr, longjmp, sizeof (jmp_buf))))
+    test = setjmp(*png_set_longjmp_fn(png_ptr, longjmp, sizeof (jmp_buf)));
 #endif
     {
+    if (test)
```

Comment:
This should be one line above so that the if statement captures the whole block surrounded by the curly braces